Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -131,16 +135,54 @@ public interface IonSerializerFactory<T, TContext> where TContext : IonSerializa
 
     public class IonSerializer
     {
-        private readonly IonSerializationOptions options;
+        internal readonly IonSerializationOptions options;
+        private Dictionary<Type, dynamic> primitiveSerializers { get; init; }
+        private IonNullSerializer nullSerializer { get; init; }
+        private IonClobSerializer clobSerializer { get; init; }
+        private IonListSerializer listSerializer { get; init; }
+        private IonObjectSerializer objectSerializer { get; init; }
 
         public IonSerializer() : this(new IonSerializationOptions())
         {
-
         }
 
         public IonSerializer(IonSerializationOptions options)
         {
             this.options = options;
+            this.primitiveSerializers = new Dictionary<Type, dynamic>()
+            {
+                {typeof(bool), new IonBooleanSerializer()},
+                {typeof(string), new IonStringSerializer()},
+                {typeof(byte[]), new IonByteArraySerializer()},
+                {typeof(int), new IonIntSerializer()},
+                {typeof(long), new IonLongSerializer()},
+                {typeof(float), new IonFloatSerializer()},
+                {typeof(double), new IonDoubleSerializer()},
+                {typeof(decimal), new IonDecimalSerializer()},
+                {typeof(BigDecimal), new IonBigDecimalSerializer()},
+                {typeof(SymbolToken), new IonSymbolSerializer()},
+                {typeof(DateTime), new IonDateTimeSerializer()},
+                {typeof(Guid), new IonGuidSerializer(this.options.AnnotateGuids)},
+            };
+            this.nullSerializer = new IonNullSerializer();
+            this.clobSerializer = new IonClobSerializer();
+            this.listSerializer = new IonListSerializer(this);
+            this.objectSerializer = new IonObjectSerializer(this);
+
+            if (this.options.IonSerializers != null)
+            {
+                foreach (var serializer in this.options.IonSerializers)
+                {
+                    if (ValidateCustomSerializer(serializer.Key, serializer.Value))
+                    {
+                        this.primitiveSerializers[serializer.Key] = serializer.Value;

```

Comment:
No I don't think this is right. Mutating the map of primitives is a bit icky. I would rather you just look in the map at the time you need to than do this replacement.

Reply:
Addressed.